directories.go

```diff
@@ -0,0 +1,140 @@
+package main
+
```

Comment:
How about moving this into it's own (internal) package? That way, we'll have access to these functions in other internal packages, and the global `var lock` becomes scoped to these functions.

It also provides a nice namespace for the API:

```go
  func main() {
      // DataDir() becomes:
      directories.Data()

      // PrintDirectories() becomes:
      directories.Print()

     // MigrateLegacyDataDir becomes:
     directories.MigrateLegacy()
  }
```

What do you think? 🙂

Reply:
Done, but I also had to do the same with version info to continue access the AppName. Have a look.

We might need to squash this branch before merging!

Reply:
Yes, please squash and force push now 👍